#GROP ACTIVITY WEEK6  BASED ON MEASURES OF CENTRAL TENDENCY AND DISPERSION

In [10]:
import pandas as pd
countrydata='Dataset_w6.xlsx'
df=pd.read_excel(countrydata)
df


country             Region  \
0                           Afghanistan       SouthernAsia   
1                               Albania     SouthernEurope   
2                               Algeria     NorthernAfrica   
3                                Angola       MiddleAfrica   
4                   Antigua and Barbuda          Caribbean   
..                                  ...                ...   
191  Venezuela (Bolivarian Republic of)       SouthAmerica   
192                            Viet Nam  South-easternAsia   
193                               Yemen        WesternAsia   
194                              Zambia      EasternAfrica   
195                            Zimbabwe      EasternAfrica   

     Surface area (km2)  Population in thousands (2017)  \
0                652864                           35530   
1                 28748                            2930   
2               2381741                           41318   
3               1246700                           29784   
4                   442                             102   
..                  ...                             ...   
191              912050                           31977   
192              330967                           95541   
193              527968                           28250   
194              752612                           17094   
195              390757                           16530   

     GDP per capita (current US$)  Agricultural production index   \
0                           623.2                             125   
1                          3984.2                             134   
2                          4154.1                             160   
3                          4714.1                             175   
4                         14764.5                              88   
..                            ...                             ...   
191                       11068.9                             118   
192                        2067.9                             136   
193                        1106.4                             136   
194                        1311.1                             179   
195                         890.4                              99   

     International trade: Imports (million US$)  
0                                          3568  
1                                          4669  
2                                         47091  
3                                          8790  
4                                           491  
..                                          ...  
191                                       16324  
192                                      174111  
193                                        6861  
194                                        7442  
195                                        5212  

[196 rows x 7 columns]

#1 NUMBER OF COUNTRIES IN THE DATASET

In [11]:
df['country'].value_counts

<bound method IndexOpsMixin.value_counts of 0                             Afghanistan
1                                 Albania
2                                 Algeria
3                                  Angola
4                     Antigua and Barbuda
                      ...                
191    Venezuela (Bolivarian Republic of)
192                              Viet Nam
193                                 Yemen
194                                Zambia
195                              Zimbabwe
Name: country, Length: 196, dtype: object>

A total of 196 countries included in the dataset

#2 NUMBER OF COUNTRIES WITH RESPECT TO REGION

In [7]:
import pandas as pd
countrydata='Dataset_w6.xlsx'
df=pd.read_excel(countrydata)
df.groupby('Region').country.count()


Region
Caribbean            16
CentralAmerica        8
CentralAsia           5
EasternAfrica        17
EasternAsia           7
EasternEurope        10
Melanesia             5
Micronesia            4
MiddleAfrica          9
NorthernAfrica        5
NorthernAmerica       4
NorthernEurope       10
Oceania               2
Polynesia             5
South-easternAsia    11
SouthAmerica         12
SouthernAfrica        5
SouthernAsia          9
SouthernEurope       12
WesternAfrica        15
WesternAsia          18
WesternEurope         7
Name: country, dtype: int64

#3 (A) REGION WITH HIGHEST PERCAPITA ON AVERAGE

In [26]:
percapita=df.groupby('Region')['GDP per capita (current US$)'].mean()
percapita.sort_values( axis=0, ascending=False, inplace=False, kind='quicksort', na_position='last', ignore_index=False, key=None)

Region
NorthernAmerica      57659.000000
WesternEurope        55387.128571
Oceania              44823.250000
NorthernEurope       42098.220000
EasternAsia          27967.614286
WesternAsia          17013.938889
Caribbean            15135.562500
SouthernEurope       14462.741667
South-easternAsia    10299.027273
EasternEurope         9292.490000
Melanesia             9261.960000
Polynesia             8686.520000
SouthAmerica          8499.775000
CentralAmerica        6348.137500
Micronesia            5008.300000
CentralAsia           4329.880000
SouthernAfrica        4191.900000
NorthernAfrica        3934.960000
MiddleAfrica          3928.833333
SouthernAsia          2968.400000
EasternAfrica         2088.852941
WesternAfrica         1013.140000
Name: GDP per capita (current US$), dtype: float64

INFERENCE:NORTHERNAMERICA HAS HIGHEST GDP PERCAPITA ON AVERAGE WITH A VALUE OF 57659

#3(B)REGION WITH HIGHEST INTERNATIONAL TRADE: IMPORTS (million US$)


In [24]:
trade=df.groupby('Region')['International trade: Imports (million US$)'].mean()
trade.sort_values( axis=0, ascending=False, inplace=False, kind='quicksort', na_position='last', ignore_index=False, key=None)

Region
NorthernAmerica      663552.250000
EasternAsia          451995.000000
WesternEurope        407498.428571
NorthernEurope       113407.200000
Oceania              112914.500000
South-easternAsia     98324.636364
EasternEurope         85243.900000
SouthernEurope        76666.666667
SouthernAsia          58199.333333
CentralAmerica        57031.500000
WesternAsia           46025.555556
NorthernAfrica        35034.400000
SouthAmerica          32885.666667
SouthernAfrica        18005.600000
CentralAsia           12063.400000
WesternAfrica          4898.333333
EasternAfrica          4726.176471
MiddleAfrica           4381.222222
Caribbean              3057.312500
Melanesia              2896.400000
Polynesia               432.000000
Micronesia               99.000000
Name: International trade: Imports (million US$), dtype: float64

INFERENCE:NORTHERNAMERICA HAS HIGHEST INTERNATIONAL TRADE IMPORTS ON AVERAGE WITH A VALUE OF 663552.25

#4 REGION MORE CONSISTENT WITH RESPECT TO GDP PER CAPITA (current US$)


In [38]:
consregion=df.groupby('Region')['GDP per capita (current US$)']
a=consregion.std()

In [41]:
a.sort_values( axis=0, ascending=True, inplace=False, kind='quicksort', na_position='last', ignore_index=False, key=None)

Region
WesternAfrica          815.095399
NorthernAfrica         974.706575
SouthernAfrica        2177.551120
SouthernAsia          2855.708179
EasternAfrica         3716.131770
CentralAsia           4152.215490
CentralAmerica        4180.440883
SouthAmerica          4273.707394
Micronesia            4820.669694
MiddleAfrica          5274.846281
EasternEurope         5415.488739
Polynesia             6956.250286
Oceania               9233.329638
SouthernEurope        9541.703216
Melanesia            13855.842686
Caribbean            14579.371326
South-easternAsia    16311.810656
WesternAsia          18812.151459
NorthernEurope       20671.030185
WesternEurope        24770.467728
NorthernAmerica      25749.619445
EasternAsia          27485.120923
Name: GDP per capita (current US$), dtype: float64

INFERENCE:WESTERN AFRICA HAS LEAST STANDARD DEVIATION.SO WESTERN AFRICA IS THE MOST CONSISTENT REGION WITH RESPECT TO GDP PER CAPITA